test/reducers/signalRReducerTest.js

```diff
@@ -0,0 +1,214 @@
+
+        expect(result.pendingMessages).to.equal(0)
+      })
+    })
+
+    describe('From state with one message pending', function () {
+      const messageStatusOnePending = {
+        pendingMessages: 1
+      }
+      it('Should return with 1 more pending message when passed RECEIVE_MESSAGE', function () {
+        const result = signalRReducer.messageStatusReducer(messageStatusOnePending, { type: signalRActions.RECEIVE_MESSAGE })
+
+        expect(result.pendingMessages).to.equal(2)
+      })
+
+      it('Should return with 0 pending message when passed ACKNOWLEDGE_MESSAGES', function () {
+        const result = signalRReducer.messageStatusReducer(messageStatusOnePending, { type: signalRActions.ACKNOWLEDGE_MESSAGES })
+
+        expect(result.pendingMessages).to.equal(0)
+      })
+    })
+
+    it('Should return state when action has an uncaught type', function () {
+      const state = {}
+      const result = signalRReducer.messageStatusReducer(state, { type: 'IGNOREME' })
+
+      expect(Object.is(result, state)).to.be.true
+    })
+  })
+
+  describe('Filter Preferences Reducer', function () {
+    context('when action is UPDATE_FILTER_PREFERENCE_EVENTS', function () {
+      context('from default state', function () {
+        const defaultState = {
+          eventFilterType: 'sync'
+        }
+
+        const getAction = (filterType) => ({
+          type: signalRActions.UPDATE_FILTER_PREFERENCE_EVENTS,
+          filterType
+        })
+
+        describe('when action.filterType is the same as the existing filter type', function () {
+          const result = signalRReducer.filterPreferencesReducer(null)(defaultState, getAction('sync'))
+
+          it('should return state', function () {
+            expect(Object.is(result, defaultState)).to.be.true
+          })
+        })
+
+        describe('when action.filterType is an invalid filter type', function () {
+          const result = signalRReducer.filterPreferencesReducer(null)(defaultState, getAction('pool'))
+
+          it('should return state', function () {
+            expect(Object.is(result, defaultState)).to.be.true
+          })
+        })
+
+        describe('when action.filterType is a valid type different from the existing filter type', function () {
+          const result = signalRReducer.filterPreferencesReducer(null)(defaultState, getAction('none'))
+
+          it('should return a new state object with the action\'s eventFilterType', function () {
+            expect(result.eventFilterType).to.equal('none')
+          })
+        })
+      })
+    })
+
+    it('Should return state when action has an uncaught type', function () {
+      const state = { eventFilterType: 'unimportant' }
+      const result = signalRReducer.filterPreferencesReducer(undefined)(state, { type: 'IGNOREME' })
+
+      expect(Object.is(result, state)).to.be.true
```

Comment:
We want this to be the exact same object so that we don't trigger re-renders. Deeply equivalent but different objects will trigger unnecessary re-rendering.

Reply:
Then `equal` is definitely the API we should use.